Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/BayesStretch2Test.py

```diff
@@ -0,0 +1,331 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+
+from mantid.simpleapi import Load, DeleteWorkspace, CreateWorkspace, CompareWorkspaces, GroupWorkspaces
+import numpy as np
+from mantid import AnalysisDataService
+from BayesStretch2 import BayesStretch2
+
+from unittest import mock
+from quickBayes.functions.qse_fixed import QSEFixFunction
+from quickBayes.workflow.qse_search import QSEGridSearch
+
+
+SAMPLE_NAME = "__BayesStretchTest_Sample"
+RES_NAME = "__BayesStretchTest_Resolution"
+
+
+def add_log_mock(workspace, sample_logs, data_ws):
+    return workspace
+
+
+class BayesStretch2Test(unittest.TestCase):
+    """
+    These tests are for checking the quickBayes
+    lib is used correctly. The results from the
+    lib are assumed to be correct (its fully tested).
+
+    Going to test each method in isolation.
+    """
+
+    def setUp(self):
+        self._res_ws = Load(Filename="irs26173_graphite002_res.nxs", OutputWorkspace=RES_NAME)
+        self._sample_ws = Load(Filename="irs26176_graphite002_red.nxs", OutputWorkspace=SAMPLE_NAME)
+        self._alg = BayesStretch2()
+        self._alg.initialize()
+        self._N_hist = 1
```

Comment:
If we are not changing the resolution and sample test data in our tests, I think this can become a `setUpClass(cls)` method, meaning that we only need to load it in once and therefore save lots of time when the test is run